packages/nextjs/services/database/repositories/users.ts

```diff
@@ -84,6 +85,68 @@ export async function getSortedUsersWithChallengesInfo(start: number, size: numb
   };
 }
 
+export async function getSortedBatchBuilders({
+  start,
+  size,
+  sorting,
+  filter,
+  batchId,
+}: {
+  start: number;
+  size: number;
+  sorting: SortingState;
+  filter?: string;
+  batchId?: number;
+}) {
+  const sortingQuery = sorting[0] as ColumnSort;
+  const query = db.query.users.findMany({
+    limit: size,
+    offset: start,
+    with: {
+      batch: true,
+    },
+    where: users =>
+      and(
+        isNotNull(users.batchId),
+        filter ? ilike(users.userAddress, `%${filter}%`) : undefined,
```

Comment:
Future: We would be able to improve this if we do #45 , and we could also search in socials.